src/Development/IDE/Core/Rules.hs

```diff
@@ -623,12 +645,25 @@ getModIfaceFromDiskRule = defineEarlyCutoff $ \GetModIfaceFromDisk f -> do
               pure (Nothing, ([], Nothing))
 
 getModSummaryRule :: Rules ()
-getModSummaryRule = define $ \GetModSummary f -> do
+getModSummaryRule = defineEarlyCutoff $ \GetModSummary f -> do
     dflags <- hsc_dflags . hscEnv <$> use_ GhcSession f
     (_, mFileContent) <- getFileContents f
     modS <- liftIO $ evalWithDynFlags dflags $ runExceptT $
         getModSummaryFromImports (fromNormalizedFilePath f) (textToStringBuffer <$> mFileContent)
-    return $ either (,Nothing) (([], ) . Just) modS
+    case modS of
+        Right ms@ModSummary{..} -> do
+            let fingerPrint =
+                    ( moduleNameString (moduleName ms_mod)
```

Comment:
What are you basing this on? I’m slightly worried that we are missing some fields of  `ModSummary` that might be important.

Reply:
This is based on the fields of `ModSummary` as populated by `getModSummaryFromImports`. Can extract to a helper function and document its purpose and scope if it helps?